broken/tabletop/aggregate/aggregate_config.txt

// Mote and View_size specifications are relative to the x- and y-axes
// described in Image_bounds.  The correct aspect ratio of the source image
// is always maintained when it's displayed.

Title           = Haunted Map
Image bounds    = xmin 0   ymin 0   width 1   height 1
Image filename  = Berkeley_Campus_Aerial_V3_maponly_rot.jpg

View size       = width 0.500  height 0.667

Do overlay      = yes
Do mark mote    = yes

Mote = id 0x234  xpos 0      ypos 0
Mote = id 0x205  xpos 0.333  ypos 0
Mote = id 0x216  xpos 0.667  ypos 0
Mote = id 0x217  xpos 1      ypos 0
              
Mote = id 0x224  xpos 0      ypos 0.5
Mote = id 0x226  xpos 0.302  ypos 0.284
Mote = id 0x236  xpos 0.765  ypos 0.700
Mote = id 0x215  xpos 1      ypos 0.5
              
Mote = id 0x233  xpos 0      ypos 1
Mote = id 0x214  xpos 0.333  ypos 1
Mote = id 0x204  xpos 0.667  ypos 1
Mote = id 0x237  xpos 1      ypos 1



// Advanced options

Background color              = 0.5    0.5    0.5
                             
Overlay bounds                = xmin 0   ymin 0   width 1   height 1
Overlay shrink                = width 2.5  height 2.5
Overlay offset                = width 0.15   height 0.25
Overlay mote radius           = 0.12
Overlay meter size            = 1.5
Overlay background color      = 0.75   0.75   0      0.5
Overlay mote color            = 0.15   0.4    0      0.75
Overlay meter dim color       = 0      0      0      0.75
Overlay meter radio color     = 1      0.5    0
Overlay meter sense color     = 1      0      0

Overlay estimate radius       = 0.08
Overlay estimate inner radius = 0.05
Overlay estimate color        = 0.1    0.1    0.3   0.75
Overlay estimate inner color  = 0.7    0.8    1.0   0.75

Sense timeout                 = 0.2
Radio bar time                = 5
                             
Mark mote radius              = 0.02
Mark mote color               = 0.15   0.4    0      0.75
                             
Smoothing time constant       = 0.25
                             
Initial applet bounds         = xmin -4   ymin -4   width 1032   height 746



tinyos-1.x/contrib/SystemC/Config/Config_README.txt
Config README
Written by Cory Sharp
$Id: Config_README.txt,v 1.1 2003/10/09 01:35:22 cssharp Exp $


OVERVIEW

Config provides automatic code generation for configuration parameters
that can be set remotely.

In a NesC code module, create a configuration parameter with this
kind of special comment

  //!! Config [num] { [type] [name] = [init]; }

[num] is a unique number between 0 and 255 used in the bottom messaging
layer for decoding a config message.

[type] specifies this parameters C type.  If you want to use structs,
unions, or enums, typedef them in a separate header file and use that
typedef.

[name] is a unique name for the parameter.

[init] is an initial default value for the parameter.


INTERFACES AND MODULES

All interfaces and modules are placed in the build/ directory.

  Config.h
  ConfigC.nc
  ConfigM.nc
  Config_[name].nc

Config.h defines a global structure G_Config that has one field for each
Config specification.  This is where configuration values can and should
be accessed in general.

ConfigC.nc wires everything necessary for the Configuration module.  It
provides all Config_[name] interfaces.  Be sure to wire
ConfigC.StdControl to Main in your main application configuration.  Any
module that uses a Config_[name] interface will have the corresponding
config module automatically wired in ConfigC.nc.  DO NOT wire the
Config_[name] interface in your own configuration file.

ConfigM.nc is the module that does all the marshalling, unmarshalling,
and event notification for each parameter.  You need not directly
interact with this module.

Config_[name].nc is an interface created specifically for each
configuration parameter.  It looks like this

  interface Config_[name]
  {
    event void updated();
    command result_t set( [type] [name] );
  }

The updated event is fired when the configuration value has changed
either from a routing message or from a local set command.  Use the set
command to invoke an updated event -- this can be used to abstract
intermodule behaviors.  To set a config value without causing an updated
event, set G_Config.[name] directly.  It's your problem to make sure you
do this responsibly.

Note that you only need to use a Config_[name] interface if you need to
take immediate action such as reinitialization given an updated value.
If you code will take the correct action passively through accessing
G_Config.[name], then there's no need to use the Config_[name]
interface, simplifying your wiring.  Note that when you do use a
Config_[name] interface in your module, you MUST NOT wire it in your own
configuration, because it is automatically wired in ConfigC.nc.


EXAMPLE

In MyMagM.nc you can specify config type

//!! Config 130 { uint16_t mag_threshold = 32; }
//!! Config 131 { uint16_t mag_period = 64; }

When compiled, this creates Config.h file that defines a G_Config with
mag_threshold and mag_period fields.  G_Config.mag_threshold initializes
to 32 and G_Config.mag_period initializes to 64.

  command result_t StdControl.start()
  {
    call Timer.start( G_Config.mag_period );
    return SUCCESS;
  }

  event result_t Timer.fired()
  {
    call U16Sensor.read();
  }

  event result_t U16Sensor.readDone( uint16_t value )
  {
    if( value > G_Config.mag_threshold )
    {
      // ... do something clever and impressive ...
    }
  }

G_Config.mag_period sets the timer period for reading the magnetometer
value.  If it's changed, we're going to have to change the timer.  So,
we need to use the Config_mag_period interface so that we can reset the
timer.  Note, we won't wire Config_mag_period in MyMagC.nc because it is
automatically wired to MyMagC in ConfigC.nc.  The updated event code
could look like this:

  event void Config_mag_period.updated()
  {
    call Timer.stop();
    call Timer.start( G_Config.mag_period );
  }

G_Config.mag_threshold is only used passively in readDone.  When it's
changed, the correct behavior will take place next time readDone rolls
around.  So, we do NOT need to wire to the Config_mag_threshold behavior
there's nothing in particular we need to reinitialize on.


That's it!  Good luck!



tinyos-1.x/ChangeLog
2004-10-02 Joe Polastre <polastre@cs.berkeley.edu>
	* Changed Telos' baud rate to 57600 from 19200
	  Update your platform.properties file or remove the file
	  and allow SerialForwarder to regenerate the properties file
	  after recompiling net/tinyos/packet/Platform.java
	* TOSBase: New TOSBase implementation from Gilman Tolle.
          New implementation uses a dual buffer swap mechanism instead of a
          buffer pool of 5 buffers in the past implementation.  
          The past TOSBase would lock buffers to the point that 
          under high throughput, each buffer would gradually be 
          unusable until none of the buffers were useable and TOSBase 
          failed.  The new TOSBase works well for receiving and transmitting
          messages to and from the PC and maintains the buffer swap between
          UART and Radio components.  It doesn't use "hidden flags" 
          and is generally easier to decypher its operation.
	* CC2420RadioM reliability and feature updates
	  - Added RadioCoordinator events on send and receive for
	    time synchronization protocols.  Only the startSymbol()
	    event is signalled since the packet-based radio does not
	    have byte() events.  The time from the hardware capture
	    register (16-bit value from a 32768Hz clock) is put into
	    the TX and RX buffer "time" field.
	  - Reliability fixes.  Stop CC2420 from hanging on transmit
	    by flushing the radio.  On receive, look for corrupt packets
	    and lock the receive buffer while it is in use so that the
	    radio can't stomp on its own buffer thereby corrupting the
	    packet.  Fixed the checks for RXFIFO overflow and fixed the
	    RXFIFO flush operation to occur twice.
	  - Known bugs still exist in the implementation under heavy loads.
	    See CC2420RadioM comments for more information.

2004-08-13 Phil Levis <pal@cs.berkeley.edu>
        * Patched system/TimerM to fix two corner cases that could
          cause single shot timers to fail silently and make timers
          fire wildly (255ms) late. For safety reasons, users can
          no longer request continuous timers with period smaller than
          3ms.
        * Fixed a situation in the CC1000RadioIntM SPI handler where
          a failed task post could cause the system to system to
          enter livelock servicing SPI interrupts. 

2004-07-09 Alec Woo <awoo@cs.berkeley.edu>

	* tos/lib/MintRoute/MultihopWMEWMA.nc
	Fixes to a 16-bit MAC address issue that makes
	MintRoute to fail silently.

2004-06-14  Michael Demmer  <demmer@cs.berkeley.edu>

	* tools/java/net/tinyos/sim/plugins/RadioModelGuiPlugin.java 
	* tools/java/net/tinyos/sim/plugins/RadioModelPlugin.java 
	* tools/java/net/tinyos/sim/script/reflect/Radio.java 
	Fixes to the RadioModelPlugin and its GUI counterpart so the GUI
	accurately reflects changes that are made in scripts.

	* tos/platform/pc/GuiMsg.h 
	* tos/platform/pc/external_comm.c 
	* tos/platform/pc/external_comm.h
	Add support to TOSSIM for the event mask capability of Tython.
	Also add a new feature to the Tython/TOSSIM protocol for
	batch commands, improving the efficiency of the protocol.
	
	* tools/java/net/tinyos/sim/SimComm.java 
	* tools/java/net/tinyos/sim/SimProtocol.java 
	* tools/java/net/tinyos/sim/msg/.cvsignore 
	* tools/java/net/tinyos/sim/msg/Makefile 
	* tools/java/net/tinyos/sim/plugins/RadioModelPlugin.java 
	* tools/java/net/tinyos/sim/event/BeginBatchCommand.java 
	* tools/java/net/tinyos/sim/event/EndBatchCommand.java 
	Tython side of the support for batch commands, used in the radio
	model to improve the efficiency of updates.

2004-06-11  Michael Demmer  <demmer@cs.berkeley.edu>

	* tos/platform/pc/Nido.nc: 
	Add a -seed= command line option to specify the seed to be passed to
	srand(). If it's not specified, use the return from gettimeofday().

	* tools/java/net/tinyos/sim/AutoRun.java 
	* tools/java/net/tinyos/sim/EmpiricalModel.java 
	* tools/java/net/tinyos/sim/LossyBuilder.java 
	* tools/java/net/tinyos/sim/MoteLayoutPlugin.java 
	* tools/java/net/tinyos/sim/SimDriver.java 
	* tools/java/net/tinyos/sim/SimExec.java 
	* tools/java/net/tinyos/sim/plugins/CalamariPlugin.java 
	* tools/java/net/tinyos/sim/plugins/CentroidPlugin.java 
	* tools/java/net/tinyos/sim/pyscripts/objmover.py 
	* tools/java/net/tinyos/sim/pyscripts/spell_tython.py 
	* tools/java/net/tinyos/sim/pyscripts/swarm.py 
	* tools/java/net/tinyos/sim/script/reflect/Makefile 
	* tools/java/net/tinyos/sim/script/reflect/SimBindings.java 
	* tools/java/net/tinyos/sim/SimRandom.java 
	* tools/java/net/tinyos/sim/script/reflect/Random.java 
	Consolidate all the places within Tython/TinyViz that generate random
	numbers and change them to use a new class called SimRandom. This
	allows a single seed to be passed on the command line, enabling
	repeatable simulations. Pass the seed along to TOSSIM. Add a reflected
	object interface to expose SimRandom to scripts and change the
	existing scripts to use that.

2004-06-10  Michael Demmer  <demmer@cs.berkeley.edu>
	* tos/platform/pc/Nido.nc: 
	Move the call to initializeSockets() until after all the internal
	structures have been set up to avoid a potential race if a new
	connection comes in before we're ready for it.

	* tools/java/net/tinyos/sim/SimCommands.java
	* tools/java/net/tinyos/sim/msg/Makefile
	* tools/java/net/tinyos/sim/msg/.cvsignore
	* tools/java/net/tinyos/sim/script/reflect/Commands.java
	* tools/java/net/tinyos/sim/event/SetEventMask.java
	* tools/java/net/tinyos/sim/pyscripts/event-mask-example.py
	Add an event masking capability to TOSSIM / Tython. This allows a
	script to improve the runtime efficiency by eliminating unneeded
	event communication over the event channel. Include an example
	script demonstrating how to temporarily mask certain events and
	re-enable them later.

2004-04-23&25 Alec Woo  <awoo@cs.berkeley.edu>
	* tos/platform/mica/HPLMicC.nc
	* tos/platform/mica2/HPLMicC.nc
	* tos/platform/pc/HPLMicC.nc

tinyos-1.x/contrib/xbow/tos/lib/Xee/config.h
// Macros and definitions for variables stored in flash that survive
// code reloads and reboots.
//
// Copyright (c) 2004 by Sensicast, Inc.
// All rights including that of resale granted to Crossbow, Inc.
//
// Permission to use, copy, modify, and distribute this software and its
// documentation for any purpose, without fee, and without written
// agreement is hereby granted, provided that the above copyright
// notice, the (updated) modification history and the author appear in
// all copies of this source code.
//
// Permission is also granted to distribute this software under the
// standard BSD license as contained in the TinyOS distribution.
//
// @Author: Michael Newman
//
// $Id: config.h,v 1.1 2005/04/04 09:50:42 husq Exp $
//
#ifndef __CONFIG_H__
#define __CONFIG_H__
//
// Modification History:
//  22Mar04 MJNewman 4: Reduce max param size to 16.
//  23Feb04 MJNewman 3: remove application blocks
//  13Jan04 MJNewman 2: 0 never used for an ID.
//  24Dec03 MJNewman 1: Created.

// Parameter IDs are composed of an application ID and a parameter ID
// within the application. The composite forms a 32 bit unique ID which
// can be used to identify the parameter. These unique IDs are passed
// in subroutine calls.
//
// The purpose of having an application ID is to define who controls
// the specification of individual parameter IDs. Application IDs are
// assigned to companies that build hardware or software. They have no
// purpose other than as a naming convention to isolate who controls
// the specification.

#pragma pack(1)


//??? Temporary move to TOS file
enum {
	AM_XEECMDMSG = 0x55,
	AM_XEEDATAMSG = 0x56,
};


// This type is used to pass parameter IDs in subroutine calls. It must
// always be a legal argument to a subroutine.
typedef uint32_t AppParamID_t;


typedef uint8_t ParameterID_t;

// cell to store a CRC
typedef uint16_t ParamCRC_t;

// Build a 32 bit parameter ID combining an application and parameter
// ID.
//
// Inputs:
//	_appID		application ID
//	_paramID	parameter ID
// Returns:
//	combined AppParamID_t
#define APP_PARAM_ID(_appID, _paramID) ((AppParamID_t)(((uint32_t)_appID) << 8 | ((ParameterID_t)_paramID)))

// Recover the application portion of an AppParamID_t
//
// Inputs:
//	_appParamID	an AppParamID_t
// Returns:
//	the application ID portion
#define GET_APP_ID(_appParamID) (((uint32_t)_appParamID) >> 8)

// recover the parameter ID portion of an AppParamID_t
//
// Inputs:
//	_appParamID	an AppParamID_t
// Returns:
//	the parameter ID portion
#define GET_PARAM_ID(_appParamID) ((uint8_t)_appParamID)

// Some well known applications and parameters of those applications:

enum ApplicationID {
    // Parameters should never appear with an application of 0
    TOS_INVALID_APPLICATION = 0,
    TOS_SYSTEM = 1,			// the TinyOS operating System
    TOS_CROSSBOW = 2,			// Crossbow manufacturer parameters
    TOS_CC1000 = 3,			// CC1000 radio parameters
    TOS_CC2420 = 4,			// CC2420 radio parameters
    // Parameter flash memory is set to all ones when the flash is
    // cleared. Finding a parameter block as all ones should be
    // properly handled without reporting errors. 
    TOS_TEST_APPLICATION = (0xfffffeL),	// for use when a parameter ID has not yet been assigned
					// it is reused often and should never be used for a
					// released application.
    TOS_NO_APPLICATION = (0xffffffL),
};

// Header portion of initial memory block describing flash parameters
typedef struct {
    ParamCRC_t crc;			// CRC of this version block, includes
					// the data that follows this header
    uint8_t majorVersion;
    uint8_t minorVersion;
    uint16_t buildNumber;		// place for builder to put an ID number, ignored by code
    uint8_t bytes;			// number of bytes in the entire parameter block.
					// This byte count allows the parameter
					// block to be expanded in a
					// compatible way. The byte
					// count will never be less
					// than 1 and will typically be
					// a multiple of 16 to make
					// EEPROM writing easy for the
					// physical layers
} FlashVersionHeader_t;
enum {
    ParameterIgnoreVersion = 0xff,
};
// FlashVersionHeader_t NOTES:
//
// The majorVersion and minorVersion specify the shape of this table.
// Adding particular applications or parameters does not change the version.
//
// The majorVersion changes when an incompatible change occurs.
// The minorVersion changes when a compatible extension is made.

// Current version numbers
#define PARAM_MAJOR_VERSION 1
#define PARAM_MINOR_VERSION 1

// Data portion of flash parameters, padded to make it fit a line on
// some common flash memory types.
typedef struct {
    uint8_t data[16 - sizeof(FlashVersionHeader_t)]; // Where any expansion will show up. Unused and set to 0.
} FlashVersionData_t;


#define BASE_OF_PARAMETERS (0)

// This structure is located at BASE_OF_PARAMETERS of the flash memory.
// A major version of 0 or 0xff is an indication that no block is
// present. If the CRC is invalid no block is present and writing
// parameters is not allowed (unless the entire block is all zeros or all
// ones.
typedef struct {
    FlashVersionHeader_t vHdr;
    FlashVersionData_t d;
} FlashVersionBlock_t;


// The parameter ID is set to TOS_IGNORE_PARAMETER when a parameter should be ignored.
// This can be used to disable a setting without recovering the
// space.
enum ParameterID {
    TOS_UNUSED_PARAMETER = 0,		// 0 is never used, treated same as TOS_NO_PARAMETER
    TOS_IGNORE_PARAMETER = 0xfe,	// explicitly disabled
    TOS_NO_PARAMETER = 0xff,
};

// No more than 16 bytes of data in a single parameter value.
#define TOS_MAX_PARAM_LENGTH 16

// Storage for each parameter. No single parameter can have more than
// TOS_MAX_PARAM_LENGTH bytes of data.
//
// The initial design used an application block which was intended to
// contain both the CRC and application ID fields. This caused
// implementation issues with having to rewrite the application block
// each time a parameter block was written. It was also determined that
// the parameter blocks needed a CRC for two reasons. First the CRC
// check is done before the count is used which prevents sequencing
// through a potentially huge number of parameter blocks that are not
// really there. Second the split phase nature of reading from Flash
// makes it much easier to check the CRC early with all data present
// rather than remembering state and reading the parameter data twice,
// first to check the CRC and then to read the values.
// 
// With these issues it became clear that at least 3 parameter blocks
// per application were necessary to save any space over the simple
// design with no special application block. Thus the application block
// was removed.
typedef struct {
    ParamCRC_t crc;			// CRC of the parameter block including the data
    uint32_t applicationID : 24;
    uint32_t paramID : 8;
    uint8_t count;			// number of bytes in the data block
} ParameterHeader_t;

// Space to store the largest parameter 
typedef struct {
    ParameterHeader_t paHdr;
    uint8_t data[TOS_MAX_PARAM_LENGTH];	// block of data
} ParameterBlock_t;

////////////////////////////////////////
// Parameter IDs for TOS_SYSTEM 'application'
////////////////////////////////////////
enum systemParamID {
    TOS_MOTE_ID = 1,			// 16 bits of TinyOS mote ID
    TOS_MOTE_GROUP = 2,			// 8 bits of TinyOS group ID
    TOS_MOTE_MODEL = 3,			// 8 bits for model type see below
    TOS_MOTE_SUBMODEL = 4,		// 8 bits for sub model code see below
    TOS_MOTE_CPU_TYPE =5,		// 8 bits for CPU type code see below
    TOS_MOTE_RADIO_TYPE = 6,		// 8 bits for Radio type code see below
    TOS_MOTE_VENDOR = 7,		// 16 bits for manufacturer code see below
    TOS_MOTE_SERIAL = 8,		// 32 bits for serial number,
					// Expected to be unique within
					// vendor and model
    TOS_MOTE_CPU_OSCILLATOR_HZ = 9,	// 32 bits for CPU oscillator frequency in
					// cycles per second EX: 7372800  					
};

// Model values for TOS_MOTE_MODEL
enum modelID {
    TOS_MODEL_UNKNOWN = 1,
    TOS_MODEL_MICA1 = 2,
    TOS_MODEL_MICA2 = 3,
    TOS_MODEL_MICA2DOT = 4,
    TOS_MODEL_H900 = 5,
};

// Sub model values for TOS_MOTE_SUBMODEL
enum subModelID {
    TOS_SUBMODEL_UNKNOWN = 1,
    TOS_SUBMODEL_DEFAULT = 2,		// standard model
};
    
// Sub model values for TOS_MOTE_CPU_TYPE
enum cpuTypeID {
    TOS_CPU_TYPE_UNKNOWN = 1,
    TOS_CPU_TYPE_ATMEGA128 = 2,
};

// Sub model values for TOS_MOTE_RADIO_TYPE
enum radioTypeID {
    TOS_RADIO_TYPE_UNKNOWN = 1,
    TOS_RADIO_TYPE_CC1000 = 2,
    TOS_RADIO_TYPE_H900 = 3,
    TOS_RADIO_TYPE_CC2420 = 4,
};

// Vendor numbers for TOS_MOTE_VENDOR
enum vendorID {
    TOS_VENDOR_UNKNOWN = 1,
    TOS_VENDOR_CROSSBOW = 2,
    TOS_VENDOR_SENSICAST = 3,
};

////////////////////////////////////////
// Parameter IDs for TOS_CC1000 'application'
////////////////////////////////////////
enum cc1000ParamID{
    TOS_CC1000_TUNE_HZ = 1,		// 32 bits of frequency * 1,000,000 ex: 995,918,000
    TOS_CC1000_LOWER_HZ = 2,		// 32 bits of frequency * 1,000,000 ex: 916,000,000
    TOS_CC1000_UPPER_HZ = 3,		// 32 bits of frequency * 1,000,000 ex: 933,000,000
    TOS_CC1000_RF_POWER = 4, // 8 bit hex between 0x00 and 0xFF show radio transmit power, ref. CC1000 datasheet
    TOS_CC1000_RF_CHANNEL = 5, // 8 bit hex show radio preset channel, ref. CC1000const.h
};

////////////////////////////////////////
// Parameter IDs for TOS_CC2420 'application'
////////////////////////////////////////
enum cc2420ParamID{
    TOS_CC2420_TUNE_HZ = 1,		// 32 bits of frequency * 1,000,000 ex: 995,918,000
    TOS_CC2420_LOWER_HZ = 2,		// 32 bits of frequency * 1,000,000 ex: 916,000,000
    TOS_CC2420_UPPER_HZ = 3,		// 32 bits of frequency * 1,000,000 ex: 933,000,000
    TOS_CC2420_RF_POWER = 4, // 8 bit hex between 0x00 and 0xFF show radio transmit power, ref. CC2420 datasheet
    TOS_CC2420_RF_CHANNEL = 5, // 8 bit hex show radio preset channel, ref. CC2420const.h
};

////////////////////////////////////////
// Parameter IDs for TOS_CROSSBOW 'application'
////////////////////////////////////////
enum crossbowID {
    TOS_FACTORY_INFO1 = 1,			// 16 bytes of factory information (printable ascii)
    TOS_FACTORY_INFO2 = 2,			// 16 bytes of factory information (printable ascii)
    TOS_FACTORY_INFO3 = 3,			// 16 bytes of factory information (printable ascii)
    TOS_FACTORY_INFO4 = 4,			// 16 bytes of factory information (printable ascii)
};

// These are provided because the macro can not be directly used in the
// wiring:
//
//    AppConfigC.ConfigInt16[APP_PARAM_ID(TOS_SYSTEM,TOS_MOTE_ID)]->RecoverSystemParamsM.SystemMoteID;
//Fails, it must be
//   AppConfigC.ConfigInt16[CONFIG_MOTE_ID] -> RecoverSystemParamsM.SystemMoteID;
enum {
    CONFIG_NO_APPLICATION = APP_PARAM_ID(TOS_NO_APPLICATION,TOS_NO_PARAMETER),
    CONFIG_MOTE_ID = APP_PARAM_ID(TOS_SYSTEM,TOS_MOTE_ID),
    CONFIG_MOTE_GROUP = APP_PARAM_ID(TOS_SYSTEM,TOS_MOTE_GROUP),
    CONFIG_MOTE_MODEL = APP_PARAM_ID(TOS_SYSTEM,TOS_MOTE_MODEL),
    CONFIG_MOTE_SUBMODEL = APP_PARAM_ID(TOS_SYSTEM,TOS_MOTE_SUBMODEL),
    CONFIG_MOTE_CPU_TYPE = APP_PARAM_ID(TOS_SYSTEM,TOS_MOTE_CPU_TYPE),
    CONFIG_MOTE_RADIO_TYPE = APP_PARAM_ID(TOS_SYSTEM,TOS_MOTE_RADIO_TYPE),
    CONFIG_MOTE_VENDOR = APP_PARAM_ID(TOS_SYSTEM,TOS_MOTE_VENDOR),
    CONFIG_MOTE_SERIAL = APP_PARAM_ID(TOS_SYSTEM,TOS_MOTE_SERIAL),
    CONFIG_MOTE_CPU_OSCILLATOR_HZ = APP_PARAM_ID(TOS_SYSTEM,TOS_MOTE_CPU_OSCILLATOR_HZ),
    CONFIG_FACTORY_INFO1 = APP_PARAM_ID(TOS_CROSSBOW,TOS_FACTORY_INFO1),
    CONFIG_FACTORY_INFO2 = APP_PARAM_ID(TOS_CROSSBOW,TOS_FACTORY_INFO2),
    CONFIG_FACTORY_INFO3 = APP_PARAM_ID(TOS_CROSSBOW,TOS_FACTORY_INFO3),
    CONFIG_FACTORY_INFO4 = APP_PARAM_ID(TOS_CROSSBOW,TOS_FACTORY_INFO4),
    CONFIG_CC1000_TUNE_HZ = APP_PARAM_ID(TOS_CC1000,TOS_CC1000_TUNE_HZ),
    CONFIG_CC1000_LOWER_HZ = APP_PARAM_ID(TOS_CC1000,TOS_CC1000_LOWER_HZ),
    CONFIG_CC1000_UPPER_HZ = APP_PARAM_ID(TOS_CC1000,TOS_CC1000_UPPER_HZ),
    CONFIG_CC1000_RF_CHANNEL = APP_PARAM_ID(TOS_CC1000,TOS_CC1000_RF_CHANNEL),
    CONFIG_CC1000_RF_POWER = APP_PARAM_ID(TOS_CC1000,TOS_CC1000_RF_POWER),
    CONFIG_CC2420_RF_CHANNEL = APP_PARAM_ID(TOS_CC2420,TOS_CC2420_RF_CHANNEL),
    CONFIG_CC2420_RF_POWER = APP_PARAM_ID(TOS_CC2420,TOS_CC2420_RF_POWER),
};

enum RFCHANNEL{
	CC1K_MAX_RF_CHANNEL = 34,
};

#endif // configHedit

